test/optimization/test_quadratic_program.py

```diff
@@ -66,6 +66,144 @@ def test_name_setter(self):
         q_p.name = name
         self.assertEqual(q_p.name, name)
 
+    def assert_equal(self, x: Variable, y: Variable):
+        """asserts variable equality"""
+        self.assertEqual(x.name, y.name)
+        self.assertEqual(x.lowerbound, y.lowerbound)
+        self.assertEqual(x.upperbound, y.upperbound)
+        self.assertEqual(x.vartype, y.vartype)
+
+    def test_var_dict(self):
+        """test {binary,integer,continuous}_var_dict"""
+        q_p = QuadraticProgram()
+
+        c_count = 0
+        b_count = 0
+        i_count = 0
+
+        def verify_counts():
+            self.assertEqual(q_p.get_num_vars(), c_count + b_count + i_count)
+            self.assertEqual(q_p.get_num_continuous_vars(), c_count)
+            self.assertEqual(q_p.get_num_binary_vars(), b_count)
+            self.assertEqual(q_p.get_num_integer_vars(), i_count)
+
+        def check_dict(var_dict, offset):
+            verify_counts()
+            variables = var_dict.values()
+            for i, x in enumerate(variables):
+                y = q_p.get_variable(i + offset)
+                z = q_p.get_variable(x.name)
+                self.assert_equal(x, y)
+                self.assert_equal(x, z)
+
+        d_0 = q_p.continuous_var_dict(name='a', key_format='_{}', keys=3)
+        c_count += 3
+        check_dict(d_0, 0)
+
+        d_1 = q_p.binary_var_dict(name='b', keys=5)
+        b_count += 5
+        check_dict(d_1, len(d_0))
+
+        d_2 = q_p.integer_var_dict(key_format='_{}', keys=7, lowerbound=-4, upperbound=10)
+        i_count += 7
+        check_dict(d_2, len(d_0) + len(d_1))
+
+        d_3 = q_p.continuous_var_dict(name='a', key_format='_{}', keys=3)
+        c_count += 3
+        check_dict(d_3, len(d_0) + len(d_1) + len(d_2))
+
+        d_4 = q_p.continuous_var_dict(name='c', keys=range(3))
+        c_count += 3
+        check_dict(d_4, len(d_0) + len(d_1) + len(d_2) + len(d_3))
+
```

Comment:
I'd avoid such constructions with `len() + len() + ...`. If I want to insert a new check in between, I'd have to edit all of them.
Same for the `test_var_list`